tests/py/unit/test_material.py

```diff
@@ -25,4 +25,68 @@ def test_basis_to_json():
     assertion_utils.assert_deep_almost_equal(expected_basis_config, basis.to_json())
 
 
-# TODO: Add test to check if basis.cell is changed when lattice of material is changed, and vice versa
+def test_basis_cell_lattice_sync():
+    """Test synchronization between basis.cell and material.lattice"""
+    material = Material.create(Material.default_config)
+
+    # Change lattice vectors
+    new_vectors = [[1.0, 0.0, 0.0], [0.0, 2.0, 0.0], [0.0, 0.0, 3.0]]
+    material.set_new_lattice_vectors(*new_vectors)
+
+    # Verify basis.cell matches new lattice vectors
+    assertion_utils.assert_deep_almost_equal(new_vectors, material.basis.cell.vectors_as_array)
+
+    assertion_utils.assert_deep_almost_equal(new_vectors, material.lattice.vector_arrays)
+
+
+def test_create_empty():
+    """Test creating empty materials with different lattice parameters"""
+    # Test default parameters
+    material = Material.create_empty()
+    assert material.basis.elements.values == []
+    assert material.basis.coordinates.values == []
+    assert material.lattice.type == "CUB"
+    assert material.lattice.a == 1.0
+    assert material.lattice.b == 1.0
+    assert material.lattice.c == 1.0
+    assert material.lattice.alpha == 90.0
+    assert material.lattice.beta == 90.0
+    assert material.lattice.gamma == 90.0
+    assert material.name == "New Material"
+
+    # Test custom parameters
+    material = Material.create_empty(
+        a=2.0, b=3.0, c=4.0, alpha=80.0, beta=85.0, gamma=95.0, lattice_type="TRI", name="Custom Empty"
+    )
+    assert material.basis.elements.values == []
+    assert material.basis.coordinates.values == []
```

Comment:
Test empty should only check the emptiness.

The actual parameter values should be in test_create.

On the material level we should just check lattice and basis, name - top-level params

For lattice/basis we should have separate tests (create, etc) that check their internal parameters